pkg/prometheus/operator.go

```diff
@@ -1753,6 +1753,10 @@ func (c *Operator) selectServiceMonitors(ctx context.Context, p *monitoringv1.Pr
 					break
 				}
 			}
+
+			if err = store.AddOAuth2(ctx, sm.GetNamespace(), endpoint.OAuth2, smKey); err != nil {
+				break
+			}
```

Comment:
What would happen if I run prometheus 2.26.0 and configure oauth option in ServiceMonitor? Do we have a guard somewhere to prevent such problems? If not, we should check version here and not inject oauth config when prometheus < 2.27.0 is used.

Reply:
We have this guard here: https://github.com/prometheus-operator/prometheus-operator/pull/4170/files#diff-453ce9c721c51bf9591c2b212e9ce50157b9c40d1b211177bf502be268099656R1460

That function is used for rendering all oauth2 configuration. Should we add something more or is this guard sufficient?

Reply:
Right, I somehow didn't see that :facepalm: All good from my side!